src/test/java/com/starleken/springchannel/service/UserServiceTest.java

```diff
@@ -0,0 +1,346 @@
+        boolean isNotFound = false;
+
+        //when
+        try{
+            UserFullDto findedUser = userService.findByLogin("Starleken");
+        } catch(UserIsNotFoundException ex) {
+            isNotFound = true;
+        }
+
+        //then
+        Assertions.assertTrue(isNotFound);
+    }
+
+    @Test
+    void create_happyPath() {
+        //given
+        UserCreateDto dto = new UserCreateDto();
+        dto.setEmail("vzvzx");
+        dto.setLogin("czx");
+        dto.setPassword("bzzqa");
+        dto.setImageURL("qweqw1");
+
+        //when
+        UserFullDto userFullDto = userService.create(dto);
+
+        //then
+        Assertions.assertNotNull(userFullDto);
+        Assertions.assertNotNull(userFullDto.getId());
+    }
+
+    @Test
+    void create_whenLoginExists() {
+        //given
+        boolean isSaved = true;
+
+        UserCreateDto dto1 = new UserCreateDto();
+        dto1.setEmail("dto1");
+        dto1.setLogin("starleken");
+        dto1.setPassword("dto1");
+        dto1.setImageURL("dto1");
+
+        UserCreateDto dto2 = new UserCreateDto();
+        dto2.setEmail("dto2");
+        dto2.setLogin("starleken");
+        dto2.setPassword("dto2");
+        dto2.setImageURL("dto2");
+
+        userService.create(dto1);
+
+        //when
+        try{
+            userService.create(dto2);
+        } catch (UserCredentialsAreTakenException ex){
+            isSaved = false;
+        }
+
+        Assertions.assertFalse(isSaved);
+    }
+
+    @Test
+    void create_whenEmailExists() {
+        //given
+        boolean isSaved = true;
+
+        UserCreateDto dto1 = new UserCreateDto();
+        dto1.setEmail("starleken");
+        dto1.setLogin("dto1");
+        dto1.setPassword("dto1");
+        dto1.setImageURL("dto1");
+
+        UserCreateDto dto2 = new UserCreateDto();
+        dto2.setEmail("starleken");
+        dto2.setLogin("dto2");
+        dto2.setPassword("dto2");
+        dto2.setImageURL("dto2");
+
+        userService.create(dto1);
+
+        //when
+        try{
+            userService.create(dto2);
+        } catch (UserCredentialsAreTakenException ex){
+            isSaved = false;
+        }
+
+        Assertions.assertFalse(isSaved);
+    }
+
+    @Test
+    void update_happyPath() {
+        //given
+        String newEmail = "starleken@mail.ru";
+        String newImage = "http://newImage.jpg";
+
+        UserEntity user = generateUser();
+        UserEntity savedUser = userRepository.save(user);
+
+        UserUpdateDto dto = new UserUpdateDto();
+        dto.setId(savedUser.getId());
+        dto.setEmail(newEmail);
+        dto.setImageUrl(newImage);
+
+        //when
+        UserFullDto updatedUser = userService.update(dto);
+
+        //then
+        Assertions.assertNotNull(updatedUser);
+        Assertions.assertEquals(newEmail, updatedUser.getEmail());
+        Assertions.assertEquals(newImage, updatedUser.getImageURL());
+    }
+
+    @Test
+    void update_whenEmailExists() {
```

Comment:
Try to make // given part more readable

Example for this test:

```java
        // given
        UserEntity existingUser = userDbHelper.saveUser();
        UserUpdateDto dto = generateUserUpdateDto(existingUser.getId());
        
        dto.setEmail(existingUser.getEmail); // set taken email
        
        // when
        Assertions.assertThrows(
                UserCredentialsAreTakenException.class, 
                () -> userService.update(dto)
        );
```

Ideally you should read your tests like a book. If you have to think when you read tests - it's a sign that it's bad test

Same rules as for usual code